### DISCUSSION TITLE
Can't work out connection to laser driver board

### DISCUSSION THREAD

**Discussion by PaulC2577** [arkypita/LaserGRBL#1083] [Q&A]
Hi, I'm using an Arduino Uno and CNC Shield V3 driving a homegrown router/engraver. Bought a 2.5w laser which came with a controller board LY-BC-0018 Rev 1-A. The text on controller sockets don't make sense to me as the in's seem to be out's and there is no guidance as to where the power, laser, title connect to. The only one that is reasonably clear is the fan. Any help to this complete idiot would be gratefully received.

**Comment by StuartB4** [comment]
Hi Paul. Can you please post some images of the parts and wiring you are using, laser, laser board showing the connections and any text/markings on the board near the connection sockets.  It would be easier if we can see what you have. 
Thanks.

**Comment by StuartB4** [comment]
If this is the laser driver then it should be connected like this.
![$_31](https://user-images.githubusercontent.com/16980026/88547953-56107b00-d016-11ea-9f78-924bd3775d95.jpg)

**Reply by dvhar** [reply]
Thanks StuartB4, this is the only documentation I could find for this board. Is the PWM signal 5v? Would full power just be a constant 5v signal?

**Reply by StuartB4** [reply]
Yes. The PWM signal comes from the chip on the main controller and should be 0v for laser off, 5volts for full power. Then you get varying power between 0v and 5v. So 2.5v would be half of the full power.

**Reply by Melissa3920** [reply]
Okay, before I go and blow something up (not sure if that's a good thing or not), the connection you refer to is the ttl connector. It shares a ground with the laser diode connector, but I'm not sure what the positive goes to, or what really differentiates the ttl from the pwm signal. I'm thinking you got lucky on how you hooked this up. There is another connection over by the power input that is labelled "In+". The thing is, it is not the 12v+ pin. So, before I go... oh, I said that already... is this the "real" pwm pin that I would attach to the z+ on the cnc shield?

**Comment by StuartB4** [comment]
On the CNC shield any of the pins down the right hand side of the board are ground, so you can use The Z+ or Z- for the PWM signal and one of the ground pins next to it. So you can use a 2 pin connector. Like this. My CNC shield is a Version 3.51 so the connection for the PWM (Yellow and Black wires) come from the SpnEn pin and Ground instead of the Z+/- pins, but it's basically the same.
![IMG_2896](https://user-images.githubusercontent.com/16980026/88548682-55c4af80-d017-11ea-9adb-130f68a4fc77.JPG)

**Comment by PaulC2577** [comment]
Thanks very much for that StuartB4. That's exactly the one. The ac threw me as I only thought it would be12v dc. Also the fan "in". 
I thought I would ride in on the 'laser not working at full power when engraving 
' post, and I think you have responded to that as well. 
Hopefully, if I haven't already fried the board, should be up and running tomorrow.

**Comment by StuartB4** [comment]
Ok. Yeah AC or DC is fine. The big black component on the laser driver is a bridge rectifier. It has 4 pins, 2 for AC in and 2 for DC out. It converts any AC to DC,  but you can still apply DC without any problem because it will just go straight through the diodes inside.  
![557px-Diode_bridge_alt_2 svg](https://user-images.githubusercontent.com/16980026/88573545-d47e1480-d038-11ea-8d36-815f3385c686.png)
